server/clamav.go

```diff
@@ -50,26 +53,44 @@ func (s *Server) scanHandler(w http.ResponseWriter, r *http.Request) {
 
 	s.logger.Printf("Scanning %s %d %s", filename, contentLength, contentType)
 
-	var reader io.Reader
+	var reader io.ReadSeeker
 
-	reader = r.Body
+	body, err := ioutil.ReadAll(r.Body)
```

Comment:
This can cause OOM's very quickly with the nature that we allow 10G files.
Having two 10G files entirely read like this should blocks 20G of memory.

I guess there is not much other we can do than writing it into a temp file first to avoid bloating the memory..